compute_tools/src/compute.rs

```diff
@@ -442,8 +442,43 @@ impl ComputeNode {
 
         let pg = self.start_postgres(spec.storage_auth_token.clone())?;
 
+        // Maybe apply the spec
         if spec.spec.mode == ComputeMode::Primary {
-            self.apply_config(&compute_state)?;
+            let spec = &compute_state.pspec.as_ref().expect("spec must be set").spec;
+
+            // Get spec_id or make it up by hashing
+            //
+            // TODO Make spec_id required so there would be no need to hash.
```

Comment:
self-reminder to address before merging

Reply:
@tychoish I know we want something entirely different here long term. But for now it would be nice to avoid hashing here. I noticed that the operation_uuid is required on the console side https://github.com/neondatabase/cloud/blob/9a99ece63d4b69499cb08be0e502435e2837d732/goapp/internal/dto/zctl.go#L25

Am I right that I can make this field required without breaking anything?

Reply:
yeah operation UUID is definitely going to be populated in any valid doc. 

not all operations change the spec, though, so if the goal is to not run the spec unless we need to, this will definitely not work as we want.

Reply:
Oh I see. So for now I'll have to use hashing (or store the full spec, it's not that big), and later we can add incremental ids to save some bandwidth if needed

Reply:
I know we've gone back and forth on this, but...
- we should begin work to add an integer version field so that it can be ready sooner
- we should probably use timestamp, assuming they aren't also regenerated for every operation. (I will do this) 
- if we only generate timestamps once per-spec (and so it's really a version,) if we have easy access to previous specs, we just take make sure that successive specs are`max(now, previous+1ms)` (I can also do this,) 

Regardless, I think we have a plan for now (use timestamps or UUID), a plan for the medium term (make timestamps, better if needed or just use uuid), and the long time (version number)